packages/core/src/Scene.ts

```diff
@@ -232,6 +237,34 @@ export class Scene extends EngineObject {
     return this._rootEntities;
   }
 
+  /**
+   * Sun light source.
+   * @remarks If set this to null, scene will use the brightest directional light.
+   */
+  get sunSource(): DirectLight | null {
+    return this._sunSource;
+  }
+
+  set sunSource(light: DirectLight | null) {
+    if (light == null || light instanceof DirectLight) {
+      this._sunSource = light;
+    }
+  }
+
+  /**
+   * Shadow light source.
+   * @remarks If set this to null, shadow map will use the brightest directional light with shadowType is open.
+   */
+  get shadowSource(): DirectLight | null {
+    return this._shadowSource;
+  }
+
+  set shadowSource(light: DirectLight | null) {
```

Comment:
Don't need this, just sunSource or sunLight is enough! Every light can have shadow in the future